backend/hitas/views/apartment.py

```diff
@@ -366,38 +427,73 @@ class ApartmentViewSet(HitasModelViewSet):
     list_serializer_class = ApartmentListSerializer
     model_class = Apartment
 
+    @staticmethod
+    def select_index(table: str, pre2005: bool):
+        comparison = "<" if pre2005 else ">="
+
+        return f"""
+    SELECT
+        ROUND(
+            a.debt_free_purchase_price * current_{table}.value / NULLIF(original_{table}.value, 0)
+        ) AS max_price_{table}
+    FROM hitas_apartment AS a
+    LEFT JOIN hitas_{table} AS original_{table} ON
+        a.completion_date {comparison} '2005-01-01' AND original_{table}.month = DATE_TRUNC('month', a.completion_date)
+    LEFT JOIN hitas_{table} AS current_{table} ON
+        a.completion_date {comparison} '2005-01-01' AND current_{table}.month = DATE_TRUNC('month', NOW())
+    WHERE a.id = hitas_apartment.id
+"""
```

Comment:
Is it not possible to do this with Django ORM, or is there some benefit we gain from doing this with raw SQL?

Reply:
Unfortunately I could not find a way to do this. There's no way to to custom join with django ORM without using `extra`. Join here requires two comparisons and one of them uses `DATE_TRUNC`. I even tried to implement custom `ForeignObject` implementation. Maybe there's a way to do this with a custom implementation but that would require quite a bit of a deep dive into django's ORM's internals

Reply:
Lets leave it then, no need to waste time in trying solve what's already solved 😄